rust-runtime/aws-smithy-async/src/future/pagination_stream.rs

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Types to support stream-like operations for paginators.
+
+use crate::future::rendezvous;
+use pin_project_lite::pin_project;
+use std::fmt;
+use std::future::poll_fn;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pub mod collect;
+
+/// A wrapper around [`FnStream`].
+///
+/// This type provides the same set of methods as [`FnStream`], but that is meant to be used
+/// internally and not by external users.
+#[derive(Debug)]
+pub struct PaginationStream<Item>(FnStream<Item>);
```

Comment:
can you update these docs to be better focused on the end user? It's probably more relevant what methods they can call, not that it's a wrapper type around another type ;-)

Reply:
Absolutely. Updated docs with an example in b8f37ea.